src/devices/Hx711I2c/Hx711I2c.cs

```diff
@@ -0,0 +1,349 @@
+            switch (flag)
+            {
+                case 1: // RST is pressed
+                case 129: // Unclear - python collapses 129
+                    return 1;
+                case 2: // calibration is finished
+                    // When CAL button is pressed the LED turns orange
+                    // then user needs to wait a bit (1-2 seconds) then place calibration weight.
+                    // Use AutomaticCalibrationWeight to set weight you use for calibration.
+                    // After that orange LED flashes 3 times and turns off.
+                    // It means calibration is finished and this register will be 2.
+                    // On unsucessful calibration (i.e. nothing was placed) there is no extra flag
+                    // and the register is still 0.
+                    return 2;
+                default:
+                    // Nothing happened
+                    return 0;
+            }
+        }
+
+        /// <summary>
+        /// Gets weight reading. Tare should be called first.
+        /// </summary>
+        /// <returns></returns>
+        [Telemetry("Weight")]
+        public Mass GetWeight()
+        {
+            float rawReading = GetRawReading();
+
+            switch (GetPeelFlag())
+            {
+                case 1: // RST was pressed
+                    if (!IgnoreResetButton)
+                    {
+                        Tare();
+                        return Mass.FromGrams(0);
+                    }
+
+                    break;
+                case 2: // calibration is finished
+                    if (!IgnoreCalibrationButton)
+                    {
+                        CalibrationScale = GetAutomaticCalibrationScale();
+                    }
+
+                    break;
+            }
+
+            return Mass.FromGrams((rawReading - Offset) / CalibrationScale);
+        }
+
+        /// <summary>
+        /// Gets average raw reading.
+        /// </summary>
+        /// <param name="samples">Number of times sampling occurs</param>
+        /// <returns>Raw reading value</returns>
+        public float GetRawReading(int samples = 20)
+        {
+            int watchdog = samples * 10;
+            int sum = 0;
+            for (int i = 0; i < samples; i++)
+            {
+                uint weightGrams;
+                while (!TryReadSample(out weightGrams))
+                {
+                    watchdog--;
+                }
+
+                if (watchdog < 0)
+                {
+                    // this never happened during testing
+                    // but given original code could fail reading
+                    // that could cause infinite loop if it somehow happened so
+                    // as a prevention measure better to throw exception
+                    throw new InvalidOperationException("Getting weight failed. Getting bogus result.");
+                }
+
+                sum += (int)weightGrams;
```

Comment:
this is a bit dangerous because of potential overflow. Don't you want to use the float and make the math average every time? few more operations but no risk of overload. A bit of loss in precision but if you're sampling many times, it's because the sensor is not very precise.

Reply:
I've changed sum to 64-bit and it should not be possible anymore (SampleAveraging is 32-bit and sample is 24-bit which gives at most 56-bit so 64 should be enough) - having said that if someone uses uint.MaxValue as SampleAveraging this will take days to complete (I won't question what user wants though).